### DISCUSSION TITLE
How and when to use pybamm.ExplicitTimeIntegral()

### DISCUSSION THREAD

**Discussion by PaulKranefeld** [pybamm-team/PyBaMM#2570] [Q&A] [answered]
Hello, 
I would like to calculate the integral of a time-dependent quantity `U_terminal`. It is defined as `U_terminal = pybamm.boundary_value(phi_pe, "right")` This is basically a post processing step and therefore should not be considered by Pybamm as a constraint during the solving process of my model. I could evaluate the solution object after solving the model like `solution["U_terminal"]` and calculate the integral afterwards since `U_terminal` is only time dependent. However, I wonder if there is a way to do the calculation inside the model and get the result directly by calling the solution object like `solution["Extraced Energy"]` I have found the `pybamm.ExplicitTimeIntegral(children, initial_condition)` function ([Issue 2348](https://github.com/pybamm-team/PyBaMM/pull/2348)) but I do not understand exactly how to use it and if it is the right choice in my case. Basically I want to perform a calculation like `E_ex = pybamm.ExplicitTimeIntegral(U_terminal, ?)`. Thanks in advance for any kind of help.

**Comment by valentinsulzer** [comment, marked as answer]
In general, the way to do this is to add a variable:
```python3
int_U = pybamm.Variable("int_U")
model.rhs[int_U] = U
model.initial_conditions[int_U] = 0
```

In this case it looks like you're after the energy. You can calculate this by passing the option `{"calculate discharge energy": "true"}` and then the variable is `Discharge energy [W.h]`.

`ExplicitTimeIntegral` is still experimental at this stage and for internal use only.

**Reply by PaulKranefeld** [reply]
Thank you for your reply. Unfortunately, I use a custom model i created myself and so can not access these functions. (`"calculate discharge energy": "true"`) Then I will probably solve it for the time being via a post processing step and wait for this functionality to be implemented.
